src/osd/OSD.h

```diff
@@ -791,7 +791,7 @@ class OSDService {
   bool promote_throttle() {
     // NOTE: lockless!  we rely on the probability being a single word.
     promote_counter.attempt();
-    if ((unsigned)rand() % 1000 > promote_probability_millis.read())
+    if ((unsigned)rand() % 1001 >= promote_probability_millis.read())
```

Comment:
i think it's fine to always consider promoting an object when we have a "1000" `promote_probability_millis`. and `promote_probability_millis` is not always 1000, so the throttling still works as expected.